### PR TITLE
Check interface of Data.Foldable.Compat

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

**Comment by sol** [comment]
@RyanGlScott Can you give this a quick review?

File: src/Data/List/Compat.hs [hunk 1]
```diff
@@ -1,9 +1,11 @@
 module Data.List.Compat (
   module Base
```
Comment (sol) [inline src/Data/List/Compat.hs]: Only re-exporting `Data.List` for the latest version of `base` gives us the guarantee that the `Data.List.Compat` module is indeed exactly the same as `Data.List`.  This is a nice thing, as we can use `Data.List.Compat` with the latest version of base as a reference for testing.

File: src/Data/Foldable/Compat.hs [hunk 2]
```diff
@@ -1,15 +1,37 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 module Data.Foldable.Compat (
   module Base
```
Comment (sol) [inline src/Data/Foldable/Compat.hs]: Same, here.  Only re-exporting `Base` allows us to use the `.Compat` module with the latest version of base as reference.

File: src/Data/Foldable/Compat.hs [hunk 3]
```diff
@@ -2,14 +2,37 @@
...
```
Comment (sol) [inline src/Data/Foldable/Compat.hs]: Same, here. Only re-exporting `Base` for the latest version of `base` allows us to use the `.Compat` module with the latest version of `base` as reference.

**Comment by sol** [comment]
Next thing would be doing the same thing for `Data.List.Compat`, which is still exposes less general functions than `Data.List` from `base-4.8.0.0`.

**Comment by RyanGlScott** [comment]
If I understand this correctly, this runs a script to ensure that the type signatures of all functions exported by `Data.Foldable.Compat` always match `Data.Foldable` from `base-4.8.0.0`, regardless of the GHC version? (That is, we'd have to change `Control.Monad.Compat` and `Data.List.Compat` to use the generalized functions separately.)

Everything looks good to me.

**Comment by sol** [comment]
@RyanGlScott thx.

**Pull request merged**